src/components/App/App.component.jsx

```diff
@@ -1,57 +1,165 @@
-import React, { useLayoutEffect } from 'react';
-import { BrowserRouter, Switch, Route } from 'react-router-dom';
-
-import AuthProvider from '../../providers/Auth';
-import HomePage from '../../pages/Home';
-import LoginPage from '../../pages/Login';
-import NotFound from '../../pages/NotFound';
-import SecretPage from '../../pages/Secret';
-import Private from '../Private';
-import Fortune from '../Fortune';
-import Layout from '../Layout';
-import { random } from '../../utils/fns';
+import React, { useState, useReducer } from 'react';
+import HomePage from '../../pages/Home.page';
+import NavBar from '../NavBar';
+import useVideoApi from '../../hooks/useVideosAPI';
+import DetailsView from '../../pages/DetailsView.page';
+import Favorites from '../../pages/Favorites.page';
+import Login from '../../pages/Login.page';
+import FavoriteDetailsView from '../../pages/FavoriteDetailsView.page';
+import GlobalContext from '../../state/GlobalContext';
+import GlobalReducer from '../../reducers/GlobalReducer';
+import Drawer from '@material-ui/core/Drawer';
+import { makeStyles } from '@material-ui/core';
+import { List, ListItem, Modal } from '@material-ui/core';
+import { BrowserRouter as Router, Switch, Route, Link, Redirect } from 'react-router-dom';
+
+const initialState = {
+  theme: 'light',
+  searchText: 'wizeline',
+  drawerOpen: false,
+  user: {
+    id: '',
+    name: '',
+    loggedIn: false,
+    avatarUrl: '',
+  },
+  favouriteVideos: [],
+};
+
+const drawerWidth = 240;
+
+const useStyles = makeStyles({
+  drawer: {
+    width: drawerWidth,
+  },
+});
 
 function App() {
-  useLayoutEffect(() => {
-    const { body } = document;
+  const [state, dispatch] = useReducer(GlobalReducer, initialState);
```

Comment:
The `GlobalReducer` contains Upper case, but actually it is just a JS function, 

I suggest you to rename it as `globalReducer` in order to have consistency on the names variables (since the React components use Upper case as well).